internal/gatewayapi/listener.go

```diff
@@ -510,22 +510,25 @@ func (t *Translator) processTracing(gw *gwapiv1.Gateway, envoyproxy *egv1a1.Envo
 }
 
 func proxySamplingRate(tracing *egv1a1.ProxyTracing) float64 {
-	rate := 100.0
-	if tracing.SamplingRate != nil {
-		rate = float64(*tracing.SamplingRate)
-	} else if tracing.SamplingFraction != nil {
+	if tracing.SamplingFraction != nil {
 		numerator := float64(tracing.SamplingFraction.Numerator)
 		denominator := float64(100)
 		if tracing.SamplingFraction.Denominator != nil {
 			denominator = float64(*tracing.SamplingFraction.Denominator)
 		}
 
-		rate = numerator / denominator
+		rate := numerator / denominator
 		// Identifies a percentage, in the range [0.0, 100.0]
 		rate = math.Max(0, rate)
 		rate = math.Min(100, rate)
+		return rate
```

Comment:
Out of curiousity, this percentage is indeed specified as `[0.0, 100.0]` here: https://www.envoyproxy.io/docs/envoy/latest/api-v3/type/v3/percent.proto#envoy-v3-api-msg-type-v3-percent

Does this really mean that the values such as `0.05` would not be allowed? I don't see any code here that does something to that effect. Hence the question. Also: all of the test cases show only really trivial cases such as 1/100 or 100/100 or something.

Reply:
I think https://github.com/envoyproxy/gateway/issues/5486 will help on this.

Reply:
`0.05` is valid

Reply:
👍